tools/common-code/cmonitor_watcher.py

```diff
@@ -0,0 +1,196 @@
+    - check the process name against the white-list given in the filter list.
+    - store the events in Queue.
+    """
+
+    def __init__(self, path, filter, timeout):
+        """Initialize CgroupWatcher
+        Args:
+          path: path to watch for events.
+          filter: white-list against which the process-event is filtered.
+
+        """
+        self.path = path
+        self.filter = filter
+        self.timeout = timeout
+        self.myFileList = {}
+
+    def __get_cgroup_version(self):
+        """
+        Detect the cgroup version.
+        """
+        proc_self_mount = "/proc/self/mounts"
+        ncgroup_v1 = 0
+        ncgroup_v2 = 0
+        with open(proc_self_mount) as file:
+            for line in file:
+                row = line.split()
+                fs_spec = row[0]
+                fs_file = row[1]
+                fs_vfstype = row[2]
+                if (fs_spec == "cgroup" or fs_spec == "cgroup2") and fs_vfstype == "cgroup2":
+                    ncgroup_v2 += 1
+                else:
+                    ncgroup_v1 += 1
+
+        if ncgroup_v1 == 0 and ncgroup_v2 > 0:
+            cgroup_versopn = "v2"
+            return cgroup_version
+        else:
+            cgroup_version = "v1"
+            return cgroup_version
+
+    def __get_process_name(self, pid):
+        """Returns the process name for the process id.
+        Args:
+          pid: process id.
+
+        Returns:
+          The process name.
+
+        """
+        cgroup_version = self.__get_cgroup_version()
+        if cgroup_version == "v1":
+            proc_filename = "/proc" + "/" + pid + "/stat"
+        else:
+            proc_filename = "/proc" + "/" + pid + "/cgroup.procs"
+        with open(proc_filename) as file:
+            for line in file:
+                parts = line.split()
+                process_name = parts[1].strip("()")
+        return process_name
+
+    def __get_pid_list(self, filename):
+        """Get the list of the process ids belong to a tasks file.
+        Args:
+          filename: the tasks file.
+
+        Returns:
+          The list of PIDs within the tasks file.
+
+        """
+        list = []
+        with open(filename) as file:
+            for line in file:
+                list.append(line.strip())
+        return list
+
+    def __get_list_of_files(self, dir):
+        """Returns the list of the files created for the event within the watched dir.
+        Args:
+          filename: dir to be watched.
+
+        Returns:
+          The list of files created within the watched dir.
+
+        """
+        listOfFiles = os.listdir(dir)
+        allFiles = list()
+        for entry in listOfFiles:
+            fullpath = os.path.join(dir, entry)
+            if os.path.isdir(fullpath):
+                allFiles = allFiles + self.__get_list_of_files(fullpath)
+            else:
+                allFiles.append(fullpath)
+
+        return allFiles
+
+    def __process_task_files(self, dir):
```

Comment:
rename to __check_new_cgroup_against_filter(self, cgroup_dir) and provide an example of "cgroup_dir" that this function expects